ietf/ietfauth/views.py

```diff
@@ -113,35 +113,48 @@ def index(request):
 #     return HttpResponseRedirect(redirect_to)
 
 def create_account(request):
-    to_email = None
+    new_account_email = None
 
-    if request.method == 'POST':
+    if request.method == "POST":
         form = RegistrationForm(request.POST)
         if form.is_valid():
-            to_email = form.cleaned_data['email'] # This will be lowercase if form.is_valid()
-
-            if User.objects.filter(username=to_email).exists():
-                send_account_creation_exists_email(request, to_email)
+            new_account_email = form.cleaned_data[
+                "email"
+            ]  # This will be lowercase if form.is_valid()
+
+            user = User.objects.filter(username=new_account_email).first()
+            if user:
+                email = (
+                    new_account_email
+                    if new_account_email in user.person.email_set.filter(active=True)
+                    else user.person.email_set.filter(active=True).order_by("-time").first()
```

Comment:
Suggest ordering with `.order_by("-primary", "-time")` to prefer the primary email address

Reply:
Also, we should not mail to `new_account_email` once it has matched a `user.username` or `user.person.email_set` entry. It opens the door to unicode spoofing.

Reply:
Should we always email `user.person.email_set.filter(active=True).order_by("-primary", "-time").first()` in that case?

Reply:
Use `person.email()` instead of copying that logic everywhere please.
See
https://github.com/ietf-tools/datatracker/blob/1d87686a2bae67db0d45f6fea909c0e73856571c/ietf/person/models.py#L140-L152